tests/Doctrine/Tests/DBAL/Schema/SQLServerSchemaManagerTest.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+namespace Doctrine\Tests\DBAL\Schema;
+
+use Doctrine\DBAL\Configuration;
+use Doctrine\DBAL\Connection;
+use Doctrine\DBAL\Driver;
+use Doctrine\DBAL\Platforms\SQLServerPlatform;
+use Doctrine\DBAL\Schema\SQLServerSchemaManager;
+use PHPUnit\Framework\TestCase;
+
+/**
+ * @covers \Doctrine\DBAL\Schema\SQLServerSchemaManager
+ */
+final class SQLServerSchemaManagerTest extends TestCase
+{
+    public function testListTableNames()
+    {
+        $driverMock = $this->createMock(Driver::class);
+        $platform   = $this->createMock(SQLServerPlatform::class);
+        $connection = $this
+            ->getMockBuilder(Connection::class)
+            ->setMethods(['fetchAll'])
```

Comment:
IMO, a test like this doesn't bring much value. Mocking a DB connection means that we know for sure what a DB will return in our case which is often not true, especially given the variety of the supported platforms and their versions.

Reply:
I agree with your opinion in general.
However, this tests shows one special thing: The schema name "dbo" is getting ignored. This is done to increase compatibility to the old behaviour.

Reply:
I see your point. On the other hand, the part about not ignoring the non-`dbo` namespace is already covered by the functional test. If ignoring the `dbo` namespace is implemented in the same functional test, it will have two benefits:

1. The logic of handling `dbo` and non-`dbo` namespaces is tested in a cohesive way which make the test much more informative.
2. The test doesn't need to depend on the DB internal names. It will only test if a table was created in a non-default namespace, it will be returned with the namespace, otherwise, the namespace will be omitted. At this point, `dbo` will remain a platform-specific detail which the test will not care about.